### DISCUSSION TITLE
CoreWCF 0.1.0 GA released

### DISCUSSION THREAD

**Discussion by mconnew** [CoreWCF/CoreWCF#309] [Releases]
Release Notes available [here](https://github.com/CoreWCF/CoreWCF/releases/tag/v0.1.0).
This is the place to ask questions about the 0.1.0 release.
Blog post is now up here: https://corewcf.github.io/blog/2021/02/19/corewcf-ga-release

**Comment by Kralizek** [comment]
I'm so happy to see this! As I've been following this project since its announcement, I cannot praise enough @mconnew and everybody else involved for the great work!

I'm honestly looking forward to add WCF in my toolbox again :)

**Reply by jdillett** [reply]
Well done to everyone involved :)

**Comment by ssa3512** [comment]
Does 0.1.0 support reliable sessions with one-way net.tcp bindings?

**Reply by mconnew** [reply]
It supports one-way messages but not reliable sessions. Reliable Sessions needs a major re-write as there are some fundamental problems with how it's implemented in WCF on the server side which need to be addressed..

**Comment by tthiery** [comment]
Is the MTOM encoding working in 0.1.0? Thanks for this awesome work!

**Reply by birojnayak** [reply]
not with 0.1.0, but we have plan as per https://github.com/CoreWCF/CoreWCF/issues/234#issuecomment-703429948 . This is also needs to be done in the client side, which @mconnew is planning, afterwards we can port it to CoreWCF.

**Reply by tthiery** [reply]
Thanks. A pity, we are in need right now :)

**Reply by g7ed6e** [reply]
Client side mtom in .NET core works great with https://github.com/lennykean/WcfCoreMtomEncoder

**Reply by mconnew** [reply]
It does, but it has some major performance issues as well as functional problems. It achieves MTOM by not doing MTOM. MTOM allows any XML element to optionally be placed in a separate Multipart component of the raw request message (generally HTTP multi-part messages, but also supported in NetTcp). That encoder fakes it by saying "I support having XML elements in a separate part, I just decided not to". So it's technically by the spec, but it doesn't actually do MTOM. It's regular XML which states it could to MTOM if it wanted, it just doesn't want to. It can read MTOM, just not write it. Here's the [implementation of the write part of the MessageEncoder](https://github.com/lennykean/WcfCoreMtomEncoder/blob/master/src/WcfCoreMtomEncoder/MtomMessageEncoder.cs#L65-L73):
```
        public override ArraySegment<byte> WriteMessage(Message message, int maxMessageSize, BufferManager bufferManager, int messageOffset)
        {
            return _innerEncoder.WriteMessage(message, maxMessageSize, bufferManager, messageOffset);
        }

        public override void WriteMessage(Message message, Stream stream)
        {
            _innerEncoder.WriteMessage(message, stream);
        }
```

**Reply by mconnew** [reply]
Btw, I'm strongly for not reinventing wheels, so I had a conversation with the ASP.NET Core guys about multi-part requests/responses and I was given pointers to some reusable components to handle the multi-part aspect. This way there's less code in CoreWCF/WCF.
